### DISCUSSION TITLE
ReferenceError: dispatchReactUnityEvent

### DISCUSSION THREAD

**Discussion by Deeksha-Pandit** [jeffreylanters/react-unity-webgl#352] [Help Wanted!] [answered]
Hi, 
While trying to run the example code or even my own code, I get an error on the browser saying:
ReferenceError: dispatchReactUnityEvent is not defined _GameOver.....
I tried changing the code to window.dispatchReactUnityEvent as well but it does not work.

Unity version: 2020.3.31 and 2020.3.26
react-unity-webgl version: 8.8.0

Please let me know how I can resolve this issue?

Thank you,
Deeksha

**Comment by Deeksha-Pandit** [comment, marked as answer]
I was using the "Build and Run" option in unity to render the result and that is when I was facing this issue.
When I run it through the react application it works fine.
Closing this discussion for now.

Thank you,
Deeksha

**Reply by jeffreylanters** [reply]
Hi! Good to hear you've solved the issue. The dispatch function is not available outside of the React environment since it's injected by the React Unity WebGL module. So when using Unity's built-in Build and Run tool, these implementations are missing. 

An any fix to prevent these errors would be to wrap the dispatch call within a try catch block so it won't throw an error when trying to invoke it. 

I hope this helped you understand it a little better. Happy coding!
